File: app/src/main/java/es/etologic/mahjongscoring2/domain/use_cases/UpdateGameUseCase.java
package es.etologic.mahjongscoring2.domain.use_cases;

import javax.inject.Inject;

import es.etologic.mahjongscoring2.data.repositories.GamesRepository;
import es.etologic.mahjongscoring2.domain.model.Game;
import io.reactivex.Single;

public class UpdateGameUseCase {

    private final GamesRepository gamesRepository;

    @Inject
    public UpdateGameUseCase(GamesRepository gamesRepository) {
        this.gamesRepository = gamesRepository;
    }

    public Single<Boolean> updateGame(Game game) {
        return gamesRepository.updateOne(game)
                .flatMap(wasUpdated -> {
                    if (wasUpdated) {
                        return Single.just(Boolean.TRUE);
                    } else {
                        return Single.error(new Exception());
                    }
                });
    }
}

File: app/src/main/java/es/etologic/mahjongscoring2/domain/use_cases/UpdateGameUseCase.kt
package es.etologic.mahjongscoring2.domain.use_cases

import es.etologic.mahjongscoring2.data.repositories.GamesRepository
import es.etologic.mahjongscoring2.domain.model.Game
import io.reactivex.Single
import java.lang.Boolean.FALSE
import java.lang.Boolean.TRUE
import javax.inject.Inject

class UpdateGameUseCase @Inject
constructor(private val gamesRepository: GamesRepository) {
    
    fun updateGame(game: Game): Single<Boolean> = gamesRepository.updateOne(game)
}
